### ISSUE TITLE
ANDROID: PayloadData is always empty onNotificationClicked when app is in foreground

### ISSUE THREAD

**Issue by DoctorB** [mirzemehdi/KMPNotifier#23]
Hello, we correctly receive PayloadData in onPayloadData, but when we click on the notification in onNotificationClicked, PayloadData is always empty, do you have any idea?

Thx

Common
```
object NotificationInitializer {

    fun onApplicationStart() {
        onApplicationStartPlatformSpecific()
        NotifierManager.addListener(object : NotifierManager.Listener {
            override fun onNewToken(token: String) {
                Napier.d("Push Notification onNewToken: $token")
            }

            override fun onPushNotification(title: String?, body: String?) {
                super.onPushNotification(title, body)
                Napier.d("Push Notification notification type message is received: Title: $title and Body: $body")
            }

            override fun onPayloadData(data: PayloadData) {
                super.onPayloadData(data)
                Napier.d("Push Notification payloadData: $data")
            }

            override fun onNotificationClicked(data: PayloadData) {
                super.onNotificationClicked(data)
                Napier.d("Push Notification payloadData: $data")
            }
        })
    }
}
expect fun onApplicationStartPlatformSpecific()
```

Platform specific

```
actual fun onApplicationStartPlatformSpecific() {
    val channelId = "fcm_default_channel"

    NotifierManager.initialize(
        NotificationPlatformConfiguration.Android(
            notificationIconResId = R.drawable.ic_notifications,
            notificationIconColorResId = R.color.primaryNotification,
            notificationChannelData = NotificationPlatformConfiguration.Android.NotificationChannelData(
                id = channelId,
                name = "SmartTracker.Fcm.ChannelOreo.Name".toLocalize()
            )
        )
    )
}
```

**Comment by DoctorB** [comment]
[Notifier.kt](https://gist.github.com/DoctorB/ea3b44de8bb2b3a53f8279db9c5119d6#file-notifier-kt)
[AndroidNotifier.kt](https://gist.github.com/DoctorB/a0f3de078150d9a4f6ff1dd7e6c479ab#file-androidnotifier-kt)
[MyFirebaseMessagingService.kt](https://gist.github.com/DoctorB/9ebadba074fc5eddf9904efef5c344b2#file-myfirebasemessagingservice-kt)
[IosNotifier.kt](https://gist.github.com/DoctorB/6f6b62022bab82893e05ae03f68d594e#file-iosnotifier-kt)

**Comment by Moozart** [comment]
@DoctorB  hey, how did u applied solution? Forked a new repo or ?

**Comment by DoctorB** [comment]
Hello @Moozart , no new repository, unfortunately. I didn't have time, so I imported the library directly into my project to debug it, and I applied the changes directly on top.

**Comment by mirzemehdi** [comment]
@DoctorB for android, seems like this happens on android in foreground notification click only, right?

**Comment by mirzemehdi** [comment]
Thank you for providing solution btw @DoctorB, I'll apply in new version. Just thinking maybe without changing common Notifier interface

**Comment by DoctorB** [comment]
Hi @mirzemehdi you are right, the original payload coming from a remote or local notification was not notify.

**Comment by mirzemehdi** [comment]
This is fixed in version `0.6.0`

**Comment by jvondermarck** [comment]
@mirzemehdi Hello this problem happens to me when the app is in background and I click on the notification and inside the `onNotificationClicked` the payload data is empty.
It happens to me in Android and I did set up everything in my MainActivity.kt so I do not know why it happens...
